File: tests/domain/value_objects/barber_current_status.py
from dataclasses import dataclass
from enum import StrEnum

from domain.value_objects.base import EnumField


class BarberStatus(StrEnum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    RESTING = "resting"


@dataclass(slots=True, frozen=True)
class BarberCurrentStatus(EnumField):
    _enum_values = tuple(BarberStatus)

File: backend/App/Domain/ValueObjects/BarberCurrentStatus.php
<?php

declare(strict_types=1);

namespace App\Domain\ValueObjects;

use App\Domain\ValueObjects\Base\EnumField;

enum BarberStatus: string
{
    case Active = 'active';
    case Inactive = 'inactive';
    case Resting = 'resting';
}

final readonly class BarberCurrentStatus extends EnumField
{
    protected const string ENUM_CLASS = BarberStatus::class;
}
